src/Contacts/src/CNLabeledValue.m

```diff
@@ -22,35 +22,69 @@
 @implementation CNLabeledValue
 
 - (instancetype)initWithLabel:(NSString *)label value:(id)value {
+    if (self = [super init]) {
+//        [self applyLabeledValueVariablesIdentifier: [[NSUUID alloc] init].UUIDString label:label value:value];
+        [self applyLabeledValueVariablesIdentifier: [[NSUUID alloc] init].UUIDString label:label value:value];
+    }
+    
+    return self;
+}
+
+- (instancetype)initWithIdentifier:(NSString *)identifier label:(NSString *)label value:(id)value {
+    if (self = [super init]) {
+        [self applyLabeledValueVariablesIdentifier:identifier label:label value:value];
+    }
+    
+    return self;
+}
+
+- (void)dealloc {
+    [_identifier release];
+    [_label release];
+    [_value release];
+    [super dealloc];
+}
+
+- (instancetype)applyLabeledValueVariablesIdentifier:(NSString *)identifier label:(NSString *)label value:(id)value {
```

Comment:
There's no need for a separate method, it would be better to just put the code of this method into `initWithIdentifier:label:value:` and make `initWithLabel:value:` call that, like this:

```objc
- (instancetype) initWithLabel: (NSString *) label value: (id) value {
    NSUUID *uuid = [NSUUID UUID];
    return [self initWithIdentifier: [uuid UUIDString] label: label value: value];
}

- (instancetype) initWithIdentifier: (NSString *) identifier
                              label: (NSString *) label
                              value: (id) value {

    if (self = [super init]) {
        _identifier = [identifier copy];
        // ...
    }
    return self;
}
```